### DISCUSSION TITLE
What is your solution for SSO with Azure OIDC?

### DISCUSSION THREAD

**Discussion by x-du** [FlowiseAI/Flowise#2964] [General]
I want my chatbot to be only accessible to employees who have logged to their Azure OIDC account.. The chatbot cannot be used by public. 
What is your solution for this?  I can think of a few.  1) Change the Flowise code to add SSO with OIDC. 2) Have a API Gateway on AWS integrated with Azure OIDC. 3) Developer a proxy application.   

I'd like to hear your thought on this. Has anyone made it work? Thanks!

**Comment by yavisht** [comment]
@x-du 

If you just want to secure the api endpoint, Here are 2 ways you can do this.

First secure the chatflow api with an api key. 

### Option 1
Create an API Management (APIM) instance in the Azure portal and import the flowise chatflow api

### Option 2
Create a function app to act as a proxy for the flowise chatflow api endpoint

### Option 3
If you want to secure the entire flowise instance, you can look into Azure App Service

![image](https://github.com/user-attachments/assets/1b96bcbe-ea42-4825-bb0a-9761d5f5a0b2)

#### More info:
https://learn.microsoft.com/en-us/azure/app-service/overview-authentication-authorization


Hope this helps!
